app/assets/javascripts/components/contextual-guidance.js

```diff
@@ -0,0 +1,69 @@
+function ContextualGuidance ($module) {
+  if (!$module) $module = document
+  this.$module = $module
+  this.$fields = $module.querySelectorAll('[data-contextual-guidance]')
+}
+
+function nodeListForEach (nodes, callback) {
+  if (window.NodeList.prototype.forEach) {
```

Comment:
I think we've got the 'same' code written twice here - shall we just have the second version?

Reply:
This function is polyfilling the `forEach` for browsers that don't support it. If exists makes use of it.

Reply:
OK. So I understand that a polyfill is a good thing, but we shouldn't be putting them in our actual code - they should be defined globally. Another way of looking at it is that I could delete lines 8-10 and the code would still work...so let's delete lines 8-10 (or write the actual polyfill so it's avaialble globally.)